src/weights.jl

```diff
@@ -1,19 +1,32 @@
 
 ###### Weight vector #####
 
-immutable WeightVec{W,Vec<:RealVector}
-    values::Vec
-    sum::W
+if VERSION < v"0.6.0-dev.2123"
+    immutable WeightVec{S<:Real, T<:Real, V<:RealVector} <: RealVector{T}
+        values::V
+        sum::S
+    end
+else
+    immutable WeightVec{S<:Real, T<:Real, V<:AbstractVector{T}} <: AbstractVector{T}
+        values::V
+        sum::S
+    end
 end
 
 """
-    WeightVec(vs, [wsum])
+    function WeightVec{S<:Real, V<:RealVector}(vs::V, s::S)
 
 Construct a `WeightVec` with weight values `vs` and sum of weights `wsum`.
 If omitted, `wsum` is computed.
 """
-WeightVec{Vec<:RealVector,W<:Real}(vs::Vec,wsum::W) = WeightVec{W,Vec}(vs, wsum)
-WeightVec(vs::RealVector) = WeightVec(vs, sum(vs))
+function WeightVec{S<:Real, V<:RealVector}(vs::V, s::S)
```

Comment:
Interesting, I didn't realize that was valid syntax in 0.5 (ie: referencing an earlier argument when setting the default value).